File: app/src/main/java/com/datagene/daggerexample22/ui/AnotherActivityViewModel.java
package com.datagene.daggerexample22.ui;

import android.util.Log;

import androidx.lifecycle.ViewModel;

import java.util.Calendar;
import java.util.Date;

import javax.inject.Inject;

public class AnotherActivityViewModel extends ViewModel {
    @Inject
    public AnotherActivityViewModel(String s) {
        Log.d("zzzzzz", "another viewmodel is working" + s);
    }
}

File: app/src/main/java/com/datagene/daggerexample22/ui/AnotherActivityViewModel.kt
package com.datagene.daggerexample22.ui

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.datagene.daggerexample22.data.LocalDao
import javax.inject.Inject

class AnotherActivityViewModel @Inject constructor(localDao: LocalDao) : ViewModel() {
    companion object {
        private const val TAG = "AnotherActivityViewModel"
    }
    private val searchQuery = MutableLiveData<String>()
    val screenShots = Transformations.switchMap(searchQuery) {
        localDao.getScreenshot(it)
    }

    fun setSearchQuery(query:String){
        searchQuery.postValue(query)
    }
}